### DISCUSSION TITLE
WWDC 2024 Metal features

### DISCUSSION THREAD

**Discussion by d3-denis-zaporozhets** [KhronosGroup/MoltenVK#2256] [General]
I am wondering does Apple did some Metal improvemets for WWDC 2024? Cannot find that information.
Does something changed from the last year? Maybe some features were added and this will make conformance with Vulkan better or something like that?

Anyone know something about that?

**Comment by billhollings** [comment]
Although there doesn't seem to be any barnstormers, there were some Metal improvements announced at that should help with supporting Vulkan:

- Resource residency sets should help with bindless performance.
- Shader logging will allow us to support `GL_EXT_debug_printf`.
- New MSL math modes and precisions, might allow some shader optimization.

I haven't delved into changes to MSL itself yet.
